include/mbedtls/md.h

```diff
@@ -31,6 +31,91 @@
 #include "mbedtls/build_info.h"
 #include "mbedtls/platform_util.h"
 
+#if defined(MBEDTLS_MD_C)
+
+/*
+ * - MBEDTLS_MD_CAN_xxx is defined if the md module can perform xxx.
+ * - MBEDTLS_MD_xxx_VIA_PSA is defined if the md module performs xxx via PSA.
+ * - MBEDTLS_MD_SOME_PSA is defined if at least one algorithm is performed
+ *   via PSA.
+ * - MBEDTLS_MD_SOME_LEGACY is defined if at least one algorithm is performed
+ *   via a direct legacy call.
+ *
+ * The md module performs an algorithm via PSA if there is a PSA hash
+ * accelerator, and makes a direct legacy call otherwise.
```

Comment:
On second thoughts, I've made this overly complicated. I made it so that dispatch only goes through PSA if there's a driver for the particular algorithm. But why? If there's no driver, PSA just calls the software implementation. So this can be made a lot simpler: either all algorithms dispatch through PSA, or no algorithms dispatch through PSA.

This only impacts the code in `md.c` and a bit of the setup in `md.h`, not the modules that use md.

Reply:
[Calling `psa_hash_xxx` doesn't work](https://github.com/Mbed-TLS/mbedtls/issues/6471#issuecomment-1288918461) in two cases which are currently possible and reasonable (though they won't be possible once PSA is the only interface):

* A hash algorithm is implemented in software, but deliberately excluded from the PSA configuration (possible with `MBEDTLS_PSA_CRYPTO_CONFIG`) because algorithms based on it (e.g. RSA-PSS) are accelerated and the accelerator doesn't support that hash.
* The PSA function names make RPC calls to a server (`MBEDTLS_PSA_CRYPTO_CLIENT`), but there is also some local crypto including hashes (for example, there's hashes plus signature verification plus X.509).